src/rest_vol_file.c

```diff
@@ -755,3 +797,28 @@ RV_file_close(void *file, hid_t dxpl_id, void **req)
 
     return ret_value;
 } /* end RV_file_close() */
+
+/*-------------------------------------------------------------------------
+ * Function:    RV_iterate_copy_hid_cb
+ *
+ * Purpose:     Callback for H5Iiterate() that copies the hid_t
+ *              of each library object into a list.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              August, 2023
+ */
+herr_t
+RV_iterate_copy_hid_cb(hid_t obj_id, void *udata)
```

Comment:
This callback needs to take `max_objs` into account so that it doesn't overrun the buffer given.